src/java/com/rapleaf/jack/queries/GenericQueryBuilder.java

```diff
@@ -0,0 +1,176 @@
+package com.rapleaf.jack.queries;
+
+import java.io.IOException;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.SQLRecoverableException;
+import java.util.Set;
+
+import com.rapleaf.jack.BaseDatabaseConnection;
+import com.rapleaf.jack.Column;
+import com.rapleaf.jack.Table;
+import com.rapleaf.jack.queries.where_operators.IWhereOperator;
+
+public class GenericQueryBuilder {
+  private static int MAX_CONNECTION_RETRIES = 1;
+
+  private final GenericQuery genericQuery;
+  private final BaseDatabaseConnection dbConnection;
+
+  public GenericQueryBuilder(BaseDatabaseConnection dbConnection, GenericQuery genericQuery) {
+    this.dbConnection = dbConnection;
+    this.genericQuery = genericQuery;
+  }
+
+  public JoinConditionBuilder leftJoin(Table table) {
+    return new JoinConditionBuilder(this, JoinType.LEFT_JOIN, table);
+  }
+
+  public JoinConditionBuilder rightJoin(Table table) {
+    return new JoinConditionBuilder(this, JoinType.RIGHT_JOIN, table);
+  }
+
+  public JoinConditionBuilder innerJoin(Table table) {
+    return new JoinConditionBuilder(this, JoinType.INNER_JOIN, table);
+  }
+
+  GenericQueryBuilder addJoinCondition(JoinCondition joinCondition) {
+    genericQuery.addJoinCondition(joinCondition);
+    return this;
+  }
+
+  public GenericQueryBuilder where(Column column, IWhereOperator operator) {
```

Comment:
I don't think that's too many parens. That's roughly how many parens you'd need if you were writing the SQL too.
Having multiple arguments ANDed together sounds good. The most common queries would be ones like `where(a, b, c)` which is nice and clean.